msteams-platform/m365-apps/using-teams-client-sdk-preview.md

```diff
@@ -0,0 +1,354 @@
+
+1. Ensure you have [Teams Toolkit](https://aka.ms/teams-toolkit) `v2.10.0` or later
+1. Open the *Command palette*: `Ctrl+Shift+P`
+1. Run the command `Teams: Upgrade Teams JS SDK references to support Outlook and Office apps`
+
+Upon completion, the utility will have updated your `package.json` file with the TeamsJS SDK Preview (`@microsoft/teams-js@next`) dependency, and your `*.js/.ts` and `*.jsx/.tsx` files will be updated with:
+
+> [!div class="checklist"]
+> * `package.json` references to TeamsJS SDK Preview
+> * Import statements for TeamsJS SDK Preview
+> * [Function, Enum, and Interface calls](#apis-organized-into-capabilities) to TeamsJS SDK Preview
+> * `TODO` comment reminders to review areas that might be impacted by [Context](#updates-to-the-context-interface) interface changes
+> * `TODO` comment reminders to ensure [conversion to promises functions from callback style functions](#callbacks-converted-to-promises) has gone well at every call site the tool found
+
+> [!IMPORTANT]
+> Be sure to review all `TODO` items flagged by the tool.
+
+> [!IMPORTANT]
+> Code inside html files is not supported by the upgrade tooling and will require manual changes.
+
+## Callbacks converted to promises
+
+Teams APIs that previously took a callback parameter have been updated to return a JavaScript [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) object. These include the following APIs:
+
+```js
+app.getContext, app.initialize, appInstallDialog.openAppInstallDialog, authentication.authenticate, authentication.getAuthToken, authentication.getUser, authentication.registerAuthenticationHandlers was removed to support using Promises, calendar.openCalendarItem, calendar.composeMeeting, call.startCall, core.executeDeepLink, location.getLocation, location.showLocation, mail.openMailItem, mail.composeMail, media.captureImage, media.getMedia, media.selectMedia, media.viewImages, media.scanBarCode, meeting.getAppContentStageSharingCapabilities, meeting.getAuthenticationTokenForAnonymousUser, meeting.getIncomingClientAudioState, meeting.getLiveStreamState, meeting.getMeetingDetails, meeting.requestStartLiveStreaming, meeting.requestStopLiveStreaming, meeting.shareAppContentToStage, meeting.stopSharingAppContentToStage, meeting.toggleIncomingClientAudio, meeting.getAppContentStageSharingState, pages.backStack.navigateBack, pages.navigateCrossDomain, pages.navigateToTab, pages.tabs.getMruTabInstances, pages.tabs.getTabInstances, pages.config.setConfig, pages.config.getConfig, people.selectPeople, ChildAppWindow.postMessage, ParentAppWindow.postMessage
+```
```

Comment:
Is it possible to have this in a vertical list? It's pretty hard to digest in a long horizontal list like this. Or at least can we have it wrap?

Reply:
Zhenya and I went back and forth on this, but landed on a quick-copy widget as the easiest way for devs to grab if they want a list to scan for. We figured table format would take up too much visual space in the article (and not be of much use, esp with extra formatting)

Reply:
Okay that makes sense. Appreciate that a bunch of thought has gone into this, thanks for the info!